modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/PistacheServerCodegen.java

```diff
@@ -0,0 +1,412 @@
+        }
+        if (p instanceof MapProperty) {
+            MapProperty mp = (MapProperty) p;
+            Property inner = mp.getAdditionalProperties();
+            return getSwaggerType(p) + "<std::string, " + getTypeDeclaration(inner) + ">";
+        }
+        if (p instanceof StringProperty || p instanceof DateProperty
+                || p instanceof DateTimeProperty || p instanceof FileProperty
+                || languageSpecificPrimitives.contains(swaggerType)) {
+            return toModelName(swaggerType);
+        }
+
+        return "std::shared_ptr<" + swaggerType + ">";
+    }
+
+    @Override
+    public String toDefaultValue(Property p) {
+        if (p instanceof StringProperty) {
+            return "\"\"";
+        } else if (p instanceof BooleanProperty) {
+            return "false";
+        } else if (p instanceof DateProperty) {
+            return "\"\"";
+        } else if (p instanceof DateTimeProperty) {
+            return "\"\"";
+        } else if (p instanceof DoubleProperty) {
+            return "0.0";
+        } else if (p instanceof FloatProperty) {
+            return "0.0f";
+        } else if (p instanceof IntegerProperty || p instanceof BaseIntegerProperty) {
+            return "0";
+        } else if (p instanceof LongProperty) {
+            return "0L";
+        } else if (p instanceof DecimalProperty) {
+            return "0.0";
+        } else if (p instanceof MapProperty) {
+            MapProperty ap = (MapProperty) p;
+            String inner = getSwaggerType(ap.getAdditionalProperties());
+            return "std::map<std::string, " + inner + ">()";
+        } else if (p instanceof ArrayProperty) {
+            ArrayProperty ap = (ArrayProperty) p;
+            String inner = getSwaggerType(ap.getItems());
+            if (!languageSpecificPrimitives.contains(inner)) {
+                inner = "std::shared_ptr<" + inner + ">";
+            }
+            return "std::vector<" + inner + ">()";
+        } else if (p instanceof RefProperty) {
+            RefProperty rp = (RefProperty) p;
+            return "new " + toModelName(rp.getSimpleRef()) + "()";
+        }
+        return "nullptr";
+    }
+
+    @Override
+    public void postProcessParameter(CodegenParameter parameter) {
+        super.postProcessParameter(parameter);
+
+        boolean isPrimitiveType = parameter.isPrimitiveType == Boolean.TRUE;
+        boolean isListContainer = parameter.isListContainer == Boolean.TRUE;
+        boolean isString = parameter.isString == Boolean.TRUE;
+
+        if (!isPrimitiveType && !isListContainer && !isString && !parameter.dataType.startsWith("std::shared_ptr")) {
+            parameter.dataType = "std::shared_ptr<" + parameter.dataType + ">";
+        }
+    }
+
+    /**
+     * Location to write model files. You can use the modelPackage() as defined
+     * when the class is instantiated
+     */
+    public String modelFileFolder() {
+        return outputFolder + "/model";
```

Comment:
@sebymiano Even though Pistache does not support Windows at the moment (https://github.com/oktal/pistache/issues/6), I'll still file a change later to replace "/" with File.separator similar to what we do in other generators.